### PR TITLE
feat: introduce code style

### PR DESCRIPTION
**Pull request by canvural**
From the comment here: https://github.com/nunomaduro/larastan/pull/529#issuecomment-647034392

I just copied the config file from Pest. But if anyone has any suggestion we can change some configuration.

### CODE REVIEW

**Comment by szepeviktor** [comment]
😢 Does not catch the original reason behind my PR

https://github.com/nunomaduro/larastan/blob/8c060dfa01edeade33bde1b31e6f0b4c31bdd178/src/ApplicationResolver.php#L51

**Comment by canvural** [comment]
> Does not catch the original reason behind my PR
> 
> https://github.com/nunomaduro/larastan/blob/8c060dfa01edeade33bde1b31e6f0b4c31bdd178/src/ApplicationResolver.php#L51

Even with `global_namespace_import` turned on, it didn't work :slightly_frowning_face: 

Actually I'm thinking [Slevomat Coding Standard](https://github.com/slevomat/coding-standard) and PHP CodeSniffer would be better. What you guys think?

**Comment by szepeviktor** [comment]
> Even with `global_namespace_import` turned on, it didn't work 🙁

It could be a bug 🐞 PhpCsFixer may ignore typehints.

File: src/Properties/ModelProperty.php [hunk 1]
```diff
@@ -26,9 +26,9 @@ class ModelProperty implements PropertyReflection
     public function __construct(ClassReflection $declaringClass, Type $readableType, Type $writableType, bool $writeable = true)
     {
         $this->declaringClass = $declaringClass;
-        $this->readableType = $readableType;
-        $this->writableType = $writableType;
-        $this->writeable = $writeable;
+        $this->readableType   = $readableType;
+        $this->writableType   = $writableType;
+        $this->writeable      = $writeable;
```
Comment (mfn) [inline src/Properties/ModelProperty.php:31]: Add a new line with a variable which is longer and the whole block gets reformatted again?

(seen this in other places too in the PR)

👎 on that, IMHO unnecessary for any diff and should be disabled.

Comment (nunomaduro) [reply src/Properties/ModelProperty.php:31]: Personally I am not fan of this rule either.

**Comment by canvural** [comment]
Does anyone have any idea why `php-cs-fixer` finds errors when CI runs with `COMPOSER_VERSION=^2.0` whereas when `COMPOSER_VERSION=^1.0` used everything is fine?

**Comment by nunomaduro** [comment]
cc @GrahamCampbell

**Comment by GrahamCampbell** [comment]
Why not just properly configure StyleCI? The fact that this PR exists is evidence to the time devs waste on code style.

**Comment by szepeviktor** [comment]
We'd better listen to php-cs-fixer

```
   1) src/Properties/ContainerProperty.php (fully_qualified_strict_types)

      ---------- begin diff ----------

--- Original

+++ New

@@ @@

 

-    public function getReadableType(): \PHPStan\Type\Type

+    public function getReadableType(): Type

     {

         return $this->getType();

     }

 

-    public function getWritableType(): \PHPStan\Type\Type

+    public function getWritableType(): Type

@@ @@

 

-    public function isDeprecated(): \PHPStan\TrinaryLogic

+    public function isDeprecated(): TrinaryLogic

@@ @@

 

-    public function isInternal(): \PHPStan\TrinaryLogic

+    public function isInternal(): TrinaryLogic

     {

         return TrinaryLogic::createNo();

     }

 }

 

      ----------- end diff -----------
```

**Comment by mfn** [comment]
I for one prefer using tools I can also use locally

**Comment by GrahamCampbell** [comment]
StyleCI can be used locally too: github.com/StyleCI/CLI.

**Comment by GrahamCampbell** [comment]
But, the philosophy is that you are only wasting your own time fixing CS locally.

**Comment by canvural** [comment]
@nunomaduro Can you setup the StyleCI so it can create the style fix PRs automatically and also merges it automatically? According to the docs this possible, but needs to be enabled from the repository settings page.

**Comment by canvural** [comment]
ping @nunomaduro

**Comment by nunomaduro** [comment]
Screenshot 2020-07-28 at 14 50 43

Can you choose herE?

**Comment by canvural** [comment]
I guess the 3rd option. `Automatically send and merge fix pull requests`

**Comment by nunomaduro** [comment]
done

**Comment by canvural** [comment]
Ok, so lets let StyleCI handle everything. Closing this.

**Comment by GrahamCampbell** [comment]
NB laravel/framework uses the 4th option, rather than the 3rd option, so there's only one fix commit, and not a merge commit too.

**Pull request closed without merging**